### DISCUSSION TITLE
"Unsupported Target" while attempting to install neovide on OpenBSD...irritating...

### DISCUSSION THREAD

**Discussion by anoduck** [rust-skia/rust-skia#690] [General]
Twice in one day, two different projects. It must be my lucky day!

Trying to install neovide on OpenBSD, and during compiling the source for it, received:

```bash
error: failed to run custom build command for `skia-bindings v0.52.0`

Caused by:
  process didn't exit successfully: `/tmp/cargo-installwDg8IX/release/build/skia-bindings-31a3009098f1755d/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=SKIA_DEBUG

  --- stderr
  thread 'main' panicked at 'unsupported target: Target { architecture: "x86_64", vendor: "unknown", system: "openbsd", abi: None }', /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/skia-bindings-0.52.0/build_support/binaries_config.rs:118:18
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The mail "C" compiler for OpenBSD is clang, and I have ninja installed, so those shouldn't be an issue. I have had good success building rust packages from source in OpenBSD, and do not know why this project should be any different. Just wondering if there is a reason why OpenBSD is not supported.

**Comment by pragmatrix** [comment]
> do not know why this project should be any different.

If the binaries are missing, Google's [Skia library](https://skia.org/) gets built which usually is pretty difficult to get right.

> Just wondering if there is a reason why OpenBSD is not supported.

Most users of rust-skia use macOS, Linux, and Windows based OSs, so noone put an effort into it, and there is strong indication that Skia itself does not build on BSDs, see https://github.com/neovide/neovide/issues/628

Moving to discussions.

**Reply by anoduck** [reply]
This is true, and seems to be the case, since Neovide does not provide instructions for building on any of the BSDs.

Insert fake rant here --> 

> Blasted, all these innovations and leap forwards in technology, it is more than my decades old x86 desktop can handle... 

I jest, obviously. 

Well, maybe in two or three years, OpenBSD will catch up and then support for skia... 

I guess, in the least, I could get a NetBSD VM running, and attempt to compile skia on that to see if it still works. My NetBSD skills are a little rusty, been a few years, but it and OpenBSD are still pretty similar.

I don't know why I didn't think of this sooner, but NetBSD still supports the linux compatibility layer. I bet that has something to do with it.

**Reply by anoduck** [reply]
@pragmatrix Got a chance to use Neovide on someone's laptop, and the graphics were incredible. You have really gotten something to be proud of, this is an awesome library.
